src/components/dialog/dialog.component.tsx

```diff
@@ -82,7 +86,7 @@ export interface DialogProps extends ModalProps, TagProps {
   /** Padding to be set on the Dialog content */
   contentPadding?: ContentPaddingInterface;
   /** an optional array of refs to containers whose content should also be reachable by tabbing from the dialog */
-  focusableContainers?: React.MutableRefObject<HTMLElement | null>[];
+  focusableContainers?: CustomRefObject<HTMLElement | null>[];
```

Comment:
```suggestion
  focusableContainers?: CustomRefObject<HTMLElement>[];
```

the `null` option shouldn't be needed here as that's taken care of in `CustomRefObject`